extras/help_generator/generate_help.py

```diff
@@ -44,15 +44,11 @@
 helpdir = os.path.join(build_dir, "doc", "help")
 create_helpdirs(helpdir)
 
-is_MyModule_build = "MyModule" in source_dir
-
 allfiles = []
 for dirpath, dirnames, files in os.walk(source_dir):
-    is_MyModule_dir = "MyModule" in dirpath
-    if not is_MyModule_dir or (is_MyModule_dir and is_MyModule_build):
-        for f in files:
-            if f.endswith((".sli", ".cpp", ".cc", ".h", ".py")):
-                allfiles.append(os.path.join(dirpath, f))
+    for f in files:
+        if f.endswith((".sli", ".cpp", ".cc", ".h", ".py")):
+            allfiles.append(os.path.join(dirpath, f))
 
```

Comment:
You are right, I had overlooked a number of cases. But I have found a nicer formulation, will push it after some more testing.